bundles/org.openhab.binding.automower/src/main/java/org/openhab/binding/automower/internal/bridge/AutomowerBridgeHandler.java

```diff
@@ -0,0 +1,160 @@
+            }
+        }
+    }
+
+    public AutomowerBridgeHandler(Bridge bridge, HttpClient httpClient) {
+        super(bridge);
+        this.httpClient = httpClient;
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Handler disposed.");
+        if (bridge != null) {
+            stopAutomowerBridgePolling(bridge);
+            bridge = null;
+        }
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing automower bridge handler.");
+        bridgeConfiguration = getConfigAs(AutomowerBridgeConfiguration.class);
+
+        final String appKey = bridgeConfiguration.getAppKey();
+        final String userName = bridgeConfiguration.getUserName();
+        final String password = bridgeConfiguration.getPassword();
+        if (appKey == null || appKey.isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "@text/conf-error-no-app-key");
+        } else if (userName == null || userName.isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "@text/conf-error-no-username");
+        } else if (password == null || password.isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "@text/conf-error-no-password");
+        } else {
+            updateStatus(ThingStatus.UNKNOWN);
+            if (bridge == null) {
+                bridge = new AutomowerBridge(appKey, userName, password, httpClient, scheduler);
+                // bridge.setTimeout(5000);
+
+                startAutomowerBridgePolling(bridge);
+            }
+        }
+    }
+
+    private void startAutomowerBridgePolling(AutomowerBridge bridge) {
+        if (automowerBridgePollingJob == null || automowerBridgePollingJob.isCancelled()) {
+            if (bridgeConfiguration.getPollingInterval() < 1) {
+                logger.info("No valid polling interval specified. Using default value: {}s", automowerPollingIntervalS);
+            } else {
+                automowerPollingIntervalS = bridgeConfiguration.getPollingInterval();
+            }
+            automowerBridgePollingJob = scheduler.scheduleWithFixedDelay(
+                    new AutomowerBridgePollingRunnable(this, bridge), 1, automowerPollingIntervalS, TimeUnit.SECONDS);
+        }
+    }
+
+    private void stopAutomowerBridgePolling(AutomowerBridge bridge) {
+        if (automowerBridgePollingJob != null && !automowerBridgePollingJob.isCancelled()) {
+            automowerBridgePollingJob.cancel(true);
+            automowerBridgePollingJob = null;
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+
+    }
+
+    public AutomowerBridge getAutomowerBridge() {
+        return bridge;
+    }
+
+    public Optional<MowerListResult> getAutomowers() {
+        try {
+            return Optional.of(bridge.getAutomowers());
+        } catch (Exception e) {
```

Comment:
Catching `Exception` catches also all `RuntimeException`s. Better specify the concrete type.

Reply:
Makes sense